flowmachine/tests/test_raster_statistics.py

```diff
@@ -5,106 +5,92 @@
 """
 Unit tests for the RasterStatistics utilities.
 """
-import pandas as pd
 
-from unittest import TestCase
 import pytest
 
 from flowmachine.core import Table
 from flowmachine.features.raster import RasterStatistics
 
 
-class RasterStatisticsTestCase(TestCase):
+def test_computes_expected_clipping_values(get_dataframe):
     """
-    Series of tests for the RasterStatistics() utility
-    methods.
+    RasterStatistics() returns correct values when clipping vector and raster layers.
     """
+    G = "admin2pcod"
+    vector = Table(schema="geography", name="admin2")
+    r = RasterStatistics(
+        raster="population.worldpop_2015_un", vector=vector, grouping_element=G
+    )
 
-    def setUp(self):
-        """
-        Sets up test environment.
-        """
-        #
-        #  Here we use the results calculated manually using
-        #  the method `zonal_stats` which come from the
-        #  Python package `rasterio`. Those results
-        #  are generally considered correct by the team
-        #  so we will use them to compare our PostGIS
-        #  results against.
-        #
-        self.rasterio_results = pd.read_csv("tests/data/rasterio_worldpop_results.csv")
-
-    def test_computes_expected_total_value(self):
-        """
-        RasterStatistics() computes correct total value of raster.
-        """
-        r = RasterStatistics(raster="worldpop_gambia")
-        result = r.get_dataframe()
-
-        self.assertAlmostEqual(result["statistic"].iloc[0], 2083543.79777587)
-
-    def test_computes_expected_clipping_values(self):
-        """
-        RasterStatistics() returns correct values when clipping vector and raster layers.
-        """
-        G = "district_c"
-        vector = Table(schema="public", name="gambia_admin2")
+    result = get_dataframe(r)  # Should have only _one_ entry
+    assert 1 == len(result)
+    assert "524 3 07" in result.admin2pcod.values
+    assert 2500000.0 == result.set_index("admin2pcod").loc["524 3 07"][0]
```

Comment:
It's a bit of a bummer that with this simple raster we're not really testing clipping (since the result is the same as for the total value below). But again, I'm happy to defer any changes to this to a separate PR (e.g. happy to use my existing example from last week that I haven't turned into a PR yet).

Reply:
Clipping is tested in the test circles ones, which use a circle geom to clip. Could rename the test - the expected behaviour is that we only get rows where there’s overlap between the raster and the vector.

Reply:
Hmm, I see that the circles test does exercise the clipping behaviour, but I think we _also_ want to test this in the RasterStatistics class because (i) the fact that Circles uses RasterStatistics internally is an implementation detail which could in principle change in the future; (ii) if we ever decide to change or remove Circles we'd lose this test for RasterStatistics (and it's not guaranteed developers will know/remember to add it there). I suggest to use a slightly larger raster which overlaps with multiple admin regions, but again let's do this in a separate PR.